server/middlewares/validate.ts

```diff
@@ -1,12 +1,15 @@
 import { Next } from "koa";
 import { z } from "zod";
 import { ValidationError } from "@server/errors";
-import { APIContext } from "@server/types";
+import { APIContext, BaseReq } from "@server/types";
 
-export default function validate<T extends z.ZodTypeAny>(schema: T) {
-  return async function validateMiddleware(ctx: APIContext<T>, next: Next) {
+export default function validate<T extends z.ZodType<BaseReq>>(schema: T) {
+  return async function validateMiddleware(
+    ctx: APIContext<BaseReq>,
+    next: Next
+  ) {
     try {
-      ctx.input = schema.parse(ctx.request.body);
+      ctx.input = schema.parse(ctx.request);
```

Comment:
Should probably rename `input` to something more relevant like `validatedReq`.

Reply:
I'll happily take the small amount of ambiguity here and keep it as `input`, it's shorter and clear enough for folks working on the codebase every day.